sklearn/utils/_array_api.py

```diff
@@ -669,6 +669,24 @@ def _average(a, axis=None, weights=None, normalize=True, xp=None):
     return sum_ / scale
 
 
+def _median(x, axis=None, keepdims=False, xp=None):
+    xp, _ = get_namespace(x, xp=xp)
+    if hasattr(xp, "median"):
+        # `torch.median` takes the lower of the two medians when `x` has even number
+        # of elements, thus we use `torch.quantile(q=0.5)`, which gives mean of the two
+        if _is_xp_namespace(xp, "torch"):
+            return xp.quantile(x, q=0.5, dim=axis, keepdim=keepdims)
+        return xp.median(x, axis=axis, keepdims=keepdims)
+
+    if _is_xp_namespace(xp, "array-api-strict"):
+        x_np = xp.asarray(x)
+        return numpy.median(x_np, axis=axis, keepdims=keepdims)
+
+    # `median` is not included in the Array API spec, but is implemented in most
+    # array libraries, and all that we support (as of May 2025).
+    raise NotImplementedError(f"The array namespace {xp.__name__} is not supported.")
```

Comment:
I think we should let it default to the numpy conversion. No need to raise an exception.

Reply:
I see the suggestion. I don't think we have raised errors in other functions in which we convert to numpy if the function does not exist in the namespace. We could potentially raise this error, though I am not sure if we want to do that now. 

Also if we want to raise such an error I think it might be better to do it when fetching the namespace. Otherwise it would get a bit awkward to add it in every new function that we add, that is similar.

Reply:
Yes I meant let it default to all namespaces that don't have median. In our case that would just be for array-api-strict but array-api-strict is for maintaining compatibility. So if we are doing it for array-api-strict let it be that way for other namespaces that don't follow the usual conventions either.